### PR TITLE
Marshal and save embedded documents

### PR DESCRIPTION
**Pull request by markstory**
This adds the ability to marshal and save embedded documents. The features are still pretty minimal, but functional. To come in future pull requests:

* Validation for embedded documents.
* Deeply nested embedded documents.

These will likely both end up as dot separated paths, but I've not yet considered what happens in the scenario where there are deeply nested paths with missing branch elements, or how deeply nested structures expand out.

### CODE REVIEW

File: src/Association/Embedded.php [hunk 1]
```diff
@@ -11,6 +11,10 @@
  */
 abstract class Embedded
 {
+    const ONE_TO_ONE = 'oneToOne';
```
Comment (lorenzo) [inline src/Association/Embedded.php]: missing docs for the consts

File: src/Marshaller.php [hunk 2]
```diff
@@ -78,6 +101,66 @@ public function one(array $data, array $options = [])
...
+     */
+    protected function newNested($embed, $data)
+    {
+        $class = $embed->entityClass();
+        if ($embed->type() === Embedded::ONE_TO_ONE) {
+            return new $class($data);
+        }
+        if ($embed->type() === Embedded::ONE_TO_MANY) {
+            $children = [];
+            foreach ((array)$data as $row) {
```
Comment (lorenzo) [inline src/Marshaller.php]: Why is this casted to array here instead of type hinted?

**Comment by lorenzo** [comment]
This looks fantastic, great job!

File: src/Marshaller.php (hunk 2 above)
Comment (markstory) [inline src/Marshaller.php]: I don't know, the hint makes more sense now that you mention it.

**Comment by markstory** [comment]
@lorenzo Fixed up.

**Pull request merged**